src/cdcacm_serial.c

```diff
@@ -101,14 +99,13 @@ void cdc_acm_tx()
 	{
 		if(Tx_HEAD != Tx_TAIL)
 		{
-			int cnt = 0, index = Tx_TAIL;
+			int cnt = 0;
 			gpio_pin_write(gpio_dev, TXRX_LED, 0);	//turn TXRX led on
-			for (; (index != Tx_HEAD) && (cnt < BUFFER_LENGTH);cnt++) 
+			for (; (Tx_TAIL != Tx_HEAD) && (cnt < BUFFER_LENGTH*2);cnt++)
```

Comment:
weird.... how come you're doing the initialization on line 102 instead of in the for loop?